### PR TITLE
fix: replace quotes and backticks with escaped single quotes in angular

### LINKED ISSUE #795: Show component breaks Angular output when using double quotes
### I am interested in helping provide a fix!

Yes

### Which generators are impacted?

- [ ] All
- [X] Angular
- [ ] HTML
- [ ] Qwik
- [ ] React
- [ ] React-Native
- [ ] Solid
- [ ] Stencil
- [ ] Svelte
- [ ] Vue
- [ ] Web components

### Reproduction case

https://github.com/cmgriffing/mitosis-issue-examples/tree/main/apps/show

### Expected Behaviour

The output should have escaped quotations, single quotes, or escaped backticks.

eg:
```
<ng-container *ngIf="name === \"Parent\"">
or
<ng-container *ngIf="name === 'Parent'">
or
<ng-container *ngIf="name === \`Parent\`">
```

### Actual Behaviour

```
<ng-container *ngIf="name === "Parent"">
```

The quotes break Angular's templating.

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by cmgriffing**
## Description

This currently just addresses `Show`. I have not evaluated `For` or any of the other Control Flow components closely, but so far it seems like a `Show` is the main case in which this bug happens.

When using a `Show` with a `when` that references a raw string, Angular's compile breaks.

eg:

```
<Show when={foo === "bar"}>...
```

turns into:
```
template: `
  <ng-container *ngIf="foo === "bar"">...
`
```
The wrapping backticks are intentional since that is how the single file component syntax works in angular.

As you can see, the quotes would need to be escaped. However, just escaping them was problematic. Angular wouldn't compile no matter how many times I escaped the double quotes.

One fix would be to just use single quotes in the mitosis component, however this contradicts prettier's defaults, so it didn't seem ideal.

If you were to use backticks instead of double quotes, Mitosis escapes the backticks just fine, but still breaks the compile. (unexpected character)

fixes #795

### CODE REVIEW

File: packages/core/src/generators/angular.ts [hunk 1]
```diff
@@ -30,6 +30,17 @@ import { getCustomImports } from '../helpers/get-custom-imports';
...
+// double quotes and backticks are invalid in an angular component prop binding's value (not the evaluated value of a variable, just the raw value in the template)
+function escapeComponentPropValue(str: string): string {
+  let newString = str;
+
+  ["'", '"', '`'].forEach((char) => {
+    newString = newString.replaceAll(char, `\\'`);
+  });
+
+  return newString;
+}
```
[the remark this answers is not in the document]
Comment (cmgriffing) [reply packages/core/src/generators/angular.ts:42]: Would it be preferred for me to abandon this PR for now? That way it doesn't negatively impact any Pulse or other metrics?

File: packages/core/src/generators/angular.ts [hunk 2]
```diff
@@ -133,11 +144,13 @@ export const blockToAngular = (
     str += json.children.map((item) => blockToAngular(item, options, blockOptions)).join('\n');
     str += `</ng-container>`;
   } else if (json.name === 'Show') {
-    str += `<ng-container *ngIf="${stripStateAndPropsRefs(json.bindings.when?.code, {
-      contextVars,
-      outputVars,
-      domRefs,
-    })}">`;
+    str += `<ng-container *ngIf="${escapeComponentPropValue(
```
[the remark this answers is not in the document]
Comment (cmgriffing) [reply packages/core/src/generators/angular.ts:147]: Would it be preferred for me to abandon this PR for now? That way it doesn't negatively impact any Pulse or other metrics?

**Comment by samijaber** [comment]
@cmgriffing I just merged #799 which includes the new plugin. It lives here: https://github.com/BuilderIO/mitosis/blob/e570d19212d5cb745f10d0850ffd902c37adcd8b/packages/core/src/helpers/plugins/process-code.ts#L41

This is how it's used in Vue and Solid generators:

https://github.com/BuilderIO/mitosis/blob/e570d19212d5cb745f10d0850ffd902c37adcd8b/packages/core/src/generators/vue/vue.ts#L106-L131

https://github.com/BuilderIO/mitosis/blob/e570d19212d5cb745f10d0850ffd902c37adcd8b/packages/core/src/generators/solid/index.ts#L286-L295

Note that you have the flexibility of running your transform on properties/bindings/hooks/hooks-deps. In your case, I think we just want to run the quotes replacement for properties and bindings, and can ignore the other cases. Give that a shot, and let me know how it works out!

**Comment by cmgriffing** [comment]
Oh, Awesome. I will try to get into this in the next few days.

**Comment by PatrickJS** [comment]
@cmgriffing can you update this or should I have someone takeover the PR?

**Comment by samijaber** [comment]
@cmgriffing I think I'll actually try to perform the migration to the code processor plugin: it's a bit of a hefty task, and out of scope for the relatively small fix you're trying to contribute!

**Comment by cmgriffing** [comment]
Thanks. I was going to try and do it this weekend, but if its fairly involved, you can take it over.

I will close this PR but leave the Issue open. Feel free to close the issue when ready.

**Pull request closed without merging**